### PR TITLE
Switch Open Telemetry to use in_span vs start

### PR DESCRIPTION
**Pull request by jpinz**
This pull request primarily includes changes that enhance error logging and refactor the use of OpenTelemetry spans in the `dependabot` project. 
It also adds the `message` to the `DependencyFileNotFound.error-detail` property

### CODE REVIEW

**Review by abdulapopoola — APPROVED** [review]

**Comment by JamieMagee** [comment]
CI failures are due to rate limiting, not legitimate issues.

File: updater/lib/dependabot/opentelemetry.rb [hunk 1]
```diff
@@ -56,8 +54,6 @@ def self.configure
 
     sig { returns(T.nilable(::OpenTelemetry::Trace::Tracer)) }
```
Comment (JamieMagee) [inline updater/lib/dependabot/opentelemetry.rb:55]: I don't think this is `nilable` anymore.

File: updater/lib/dependabot/opentelemetry.rb [hunk 2]
```diff
@@ -25,9 +25,7 @@ def self.should_configure?
 
     sig { void }
     def self.configure
-      return unless should_configure?
-
-      puts "OpenTelemetry is enabled, configuring..."
```
Comment (JamieMagee) [inline updater/lib/dependabot/opentelemetry.rb:30]: Is it possible to retain this check? Or do the spans fail otherwise?

**Comment by JamieMagee** [comment]
For anyone reviewing this, it's easier to see the actual changes if you hide whitespace changes.

**Review by JamieMagee — APPROVED** [review]

**Review by bdragon — APPROVED** [review]
@jpinz thanks! Looks good 🚀 

As a reviewer my main point of concern was around removing the `should_configure?` guards in a few places in `Dependabot::OpenTelemetry` and what happens if `OTEL_ENABLED` is false, but I poked around and I feel good about this.

**Comment by jpinz** [comment]
> @jpinz thanks! Looks good 🚀
> 
> As a reviewer my main point of concern was around removing the `should_configure?` guards in a few places in `Dependabot::OpenTelemetry` and what happens if `OTEL_ENABLED` is false, but I poked around and I feel good about this.

I did test it while `OTEL_ENABLED` was false, and didn't notice any issues from what I could tell.

**Pull request merged**